Здесь небольшое причесывание уже извлеченных необходимых данных.

In [0]:
import pandas as pd
import numpy as np
from sklearn import preprocessing

DATA_DIR = './drive/My Drive/Colab Notebooks/VK_internship/'

In [0]:
user_artists_df = pd.read_csv(DATA_DIR+"users_artists.csv", 
                              usecols = ["user_id",	"artists", "playtime"])
artists_id = user_artists_df["artists"].unique()

In [0]:
print(len(np.unique(artists_id)))
print(np.max(artists_id))

511768
595140


Судя по всему в итоговом наборе данных есть информация о прослушивании 511768 исполнителей, при том что максимальный идентификатор равен 595140, это означает, что в данных нет информации о 84 тысячах исполнителей, но при этом есть идентификаторы.
Для представления в виде матрицы смежности это сулит лишние 84 тысячи 
пустых столбцов и строк, поэтому лучше будет перенумеровать идентификаторы для дальнейшей работы.

In [0]:
artists_id.sort()
le = preprocessing.LabelEncoder()
le.fit(artists_id)

LabelEncoder()

In [0]:
user_artists_df.loc[:,"artists"] = le.transform(user_artists_df.loc[:,"artists"])
user_artists_df.to_csv(DATA_DIR+"users_artists.csv",index=False)
del user_artists_df

In [0]:
artist_rankings = pd.read_csv(DATA_DIR+"artists_ranks.csv")
artist_rankings = artist_rankings.loc[artist_rankings["artists"].isin(artists_id)]
artist_rankings.loc[:,"artists"] = le.transform(artist_rankings.loc[:,"artists"])
artist_rankings.to_csv(DATA_DIR+"artists_ranks.csv",index=False)
del artist_rankings

In [0]:
playlists_df = pd.read_csv(DATA_DIR+"artists_in_playlists.csv", 
                           usecols = ["playlist_id",	"playcount", "artists"])
print("Количество уникальных исполниетелей в плейлистах: ", 
      len(np.unique(playlists_df["artists"])))
print("Количество исполнителей из истории прослушиваний, которые входят хотя бы в 1 плейлист: ",
      len(set(playlists_df["artists"]).intersection(set(artists_id))))
playlists_df = playlists_df.loc[playlists_df["artists"].isin(artists_id)]
playlists_df.loc[:,"artists"] = le.transform(playlists_df.loc[:,"artists"])
playlists_df.to_csv(DATA_DIR+"artists_in_playlists.csv",index=False)
del playlists_df

Количество уникальных исполниетелей в плейлистах:  64098
Количество исполнителей из истории прослушиваний, которые входят хотя бы в 1 плейлист:  63639


In [0]:
love_df = pd.read_csv(DATA_DIR+"artists_user_likes.csv")
print("Количество уникальных исполниетелей, которым поставили лайк: ", 
      len(np.unique(love_df["artists"])))
print("Количество исполнителей из истории прослушиваний, которым поставили хотя бы 1 лайк: ",
      len(set(love_df["artists"]).intersection(set(artists_id))))
love_df = love_df.loc[love_df["artists"].isin(artists_id)]
love_df.loc[:,"artists"] = le.transform(love_df.loc[:,"artists"])
love_df.to_csv(DATA_DIR+"artists_user_likes.csv",index=False)
del love_df

Количество уникальных исполниетелей, которым поставили лайк:  109495
Количество исполнителей из истории прослушиваний, которым поставили хотя бы 1 лайк:  107563


In [0]:
del artists_id